src/expr.cpp

```diff
@@ -5073,22 +5075,19 @@ llvm::Value *VectorMemberExpr::GetValue(FunctionEmitContext *ctx) const {
         // to the same logic where it's used elsewhere
         llvm::Value *elementMask = ctx->GetFullMask();
 
-        const Type *elementPtrType = NULL;
-        if (CastType<ReferenceType>(basePtrType) != NULL)
-            elementPtrType = PointerType::GetUniform(basePtrType->GetReferenceTarget());
-        else
-            elementPtrType = basePtrType->IsUniformType() ? PointerType::GetUniform(exprVectorType->GetElementType())
-                                                          : PointerType::GetVarying(exprVectorType->GetElementType());
-
+        const PointerType *ptrType = ctx->RegularizePointer(basePtrType);
```

Comment:
`elementPtrType` was incorrect before for `ReferenceType`.
It affects code generation in LoadInst. Now for reference types we correctly set alignment = 1 here https://github.com/ispc/ispc/blob/main/src/ctx.cpp#L2440. Before it used default LLVM alignment  = 32.

Reply:
Would be good to check this in the test

Reply:
Added load_unaligned.ispc lit test.